web/app/datasets/handle_signals.py

```diff
@@ -13,14 +13,105 @@
 - allow signal API to post a message to process
 - add a proper task queue (like Celery)
 """
+# Assumptions:
+# 1) Each signal will be sent to 1 API at most (underlies both table design and
+#    routing implementation.
+
+
 import os
+import datetime
+import logging
 
 import requests
+from django.conf import settings
+from django.utils import timezone
+
+from datasets.models import MessageLog
+from datasets.external.base import get_handler
+
+# -- setup logging --
+LOG_FORMAT = '%(asctime)-15s - %(name)s - %(message)s'
+logging.basicConfig(format=LOG_FORMAT, level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+# -- Datapunt internal Signalen in Amsterdam API endpoints --
+#    note: should be set by Ansible or docker-compose.yml
+SIGNALS_API_BASE = os.getenv('SIGNALS_API_BASE', 'https://acc.api.data.amsterdam.nl')
+
+
+def _get_session_with_retries():
+    """
+    Get a requests Session that will retry 5 times on a number of HTTP 500 statusses.
+    """
+    session = requests.Session()
+
+    retries = Retry(
+        total=5,
+        backoff_factor=0.1,
+        status_forcelist=[500, 502, 503, 504],
+        raise_on_status=True
+    )
+    adapter = HTTPAdapter(max_retries=retries)
+    session.mount('https://', adapter)
+    session.mount('http://', adapter)
+
+    return session
+
+
+def _batch_signals(signals):
+    """
+    Access the Signalen in Amsterdam API, retrieve signals.
+    """
+    next_page = SIGNALS_API_BASE + '/signal/'
+
+    with _get_session_with_retries() as session:
+        result = session.get(next_page)
```

Comment:
This might fail, dependent upon the result, what happens on 4XX or 5XX results

Reply:
Not quite sure what you mean - but the assumption is that logging an error in Jenkins on unrecoverable error is better than trying to recover. On some of the 500 series of errors we retry because the server could be busy.

Reply:
As mentioned before, when do you consider the situation that bad that you can't recover. Taken into account that the first thing that will be done is trying to restart the service when it has failed. Well, if the service then runs OK than I consider this a situation that the service should handle itself.

Reply:
BTW Logging these errors is always OK.

Reply:
See my comment above - we are not crashing the running service - in stead it is the import script that may crash. I can adapt the import script to exit gracefully - but that would mask errors in the system.